modules/build_c++.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+set -e
+cd osm_pbf_parser
+make
+cd ..
+
+git clone https://github.com/dfaure/PyKOpeningHours.git
+cd PyKOpeningHours
+mkdir build
+cd build
+cmake ..
+make
+# Remove these directories, their python script don't pass CI, and we don't need them anymore, anyway
+rm -rf external/share/ECM
+rm -rf external/src
+mv PyKOpeningHours.so ../..
```

Comment:
I think the right way of doing this is by adding a setup.py on your project and include it in osmose requirement.txt

Some think like this
```
git+git://github.com/dfaure/PyKOpeningHours.git@v1.33
```

Reply:
Not sure I understand. That `setup.py` file is supposed to run `cmake`, `make` etc.?

Reply:
Yes. But I never do it myself.
https://docs.python.org/3/extending/building.html
https://martinopilia.com/posts/2018/09/15/building-python-extension.html

If it is too complicated, we will do it in your way. But my point, is how we will update it on (non docker) sever.

Reply:
Ok, I implemented a [setup.py](https://github.com/dfaure/PyKOpeningHours/blob/master/setup.py) using `skbuild` (cmake wrapper). Was a bit tricky (I even submitted [a MR for skbuild](https://github.com/scikit-build/scikit-build/pull/522), but I implemented a workaround so we don't need that fix to be merged). Anyhow, now `python setup.py build` works in PyKOpeningHours.

However adding that as a requirement in osmose-backend seems to hit an issue I don't really understand.
```
Step 10/17 : RUN pip install -r requirements.txt ${PIP_INSTALL}
 ---> Running in 489932b3afd1
Collecting git+git://github.com/dfaure/PyKOpeningHours.git@v1.1 (from -r requirements.txt (line 12))
  Cloning git://github.com/dfaure/PyKOpeningHours.git (to revision v1.1) to /tmp/pip-req-build-rabkz4i8
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python /usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpp8segvp5
       cwd: /tmp/pip-req-build-rabkz4i8
  Complete output (7 lines):
  running egg_info
  creating lib/PyKOpeningHours_package.egg-info
  writing lib/PyKOpeningHours_package.egg-info/PKG-INFO
  writing dependency_links to lib/PyKOpeningHours_package.egg-info/dependency_links.txt
  writing top-level names to lib/PyKOpeningHours_package.egg-info/top_level.txt
  writing manifest file 'lib/PyKOpeningHours_package.egg-info/SOURCES.txt'
  error: package directory 'lib/PyKOpeningHours' does not exist
  ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/local/bin/python /usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpp8segvp5 Check the logs for full command output.
```

Ah, actually `pip install .` in PyKOpeningHours gives the same error. It doesn't even try to build anything, unlike `python3 setup.py build`. Is it supposed to work? Or should I ask the skbuild people?
This might indeed all be rather complicated if skbuild isn't made for this.

But I see your point about "how will we update it". I guess the alternative is to put a copy of the PyKOpeningHours code in osmose-backend, to mimick the osm_pbf_parser solution, but that's not great. Another solution would be a git submodule, if you're up for that.

Reply:
Your project does not have the standard python module layout. Move PyKOpeningHours.pyi __init__.py pykopeninghours.cpp python_qt_wrappers.cpp python_qt_wrappers.h to dir (aka module) named `PyKOpeningHours` (or maby be `pykopeninghours` to be pytonish).

Reply:
I successfully install your module using pip. That a good new ! Thank you.

Reply:
- I think you should remove the `src` dir and have `PyKOpeningHours` on the root dir (can also remove `PyKOpeningHours/lib`). In `setup.py`, package_dir should be `package_dir={'': 'PyKOpeningHours'},`. It make more sens in python.
- It look like to me `PyKOpeningHours.so` should be in `PyKOpeningHours` in dir, and `setup.py` already tell so, but not working `cmake_install_dir='PyKOpeningHours`.
- By copping `PyKOpeningHours.so` to `PyKOpeningHours`. I am able to load the python module from `PyKOpeningHours` dir, but not from outside.

Do you want a PR on this?

Reply:
- I found the last issue with the import. The empty file `__init__.py` is missing in `PyKOpeningHours` dir. Python need this to consider a dir as a module.

Reply:
Thanks for taking a look, very appreciated.

The `src` subdir (which I also don't like) seems to be necessary for skbuild to work. It looks for PyKOpeningHours (the package name) under the package_dir (src). At least it's what I thought (so before adding the `src` subdir I tried `package_dir={'': ''}` but that triggered the skbuild bug I mentioned (and fixed)). If it works for you that's great. I can make change myself or merge a MR from you, as you prefer.

I don't really understand your third point though. At but I see a weird install dir indeed, `~/lib/PyKOpeningHours/lib/PyKOpeningHours`, this makes little sense. CMake alone works. So maybe cmake_install_dir should be empty?

Does `python -c 'import PyKOpeningHours; help(PyKOpeningHours)'` work after your changes? It was obviously working in my initial MR, but since the switch to using pip, it doesn't work anymore.

Reply:
> I found the last issue with the import. The empty file __init__.py is missing in PyKOpeningHours dir. Python need this to consider a dir as a module.
 
Heh, my last commit was about removing it, thinking it would help [because help() showed `__init__.py` and nothing else]. I just reverted that commit, so it's back :)

Reply:
> * I think you should remove the `src` dir and have `PyKOpeningHours` on the root dir (can also remove `PyKOpeningHours/lib`). In `setup.py`, package_dir should be `package_dir={'': 'PyKOpeningHours'},`. It make more sens in python.

This leads to `pip install .` saying `error: package directory 'PyKOpeningHours/PyKOpeningHours' does not exist`

Reply:
I make a more simple version and I'm able to compile and install it with pip.

`~/lib/PyKOpeningHours/lib/PyKOpeningHour`, make no sense for me. It should be in`lib/python3.7/site-packages/PyKOpeningHours`. By moving at this location it is better. But now I have an issue about missing `libKOpeningHours.so.1`

Reply:
Yeah lib/python3.x is missing. Ah you're saying I should do that by hand, I see.

The .so.1 issue is solved if you rebase, it's using a static lib now.

Sorry in advance for the merge conflicts...